iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tools/cleanup/InfinispanTenantCleaner.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *  Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.enmasse.iot.tools.cleanup;
+
+import io.enmasse.iot.service.base.infinispan.cache.DeviceConnectionCacheProvider;
+import io.enmasse.iot.service.base.infinispan.cache.DeviceManagementCacheProvider;
+import io.enmasse.iot.service.base.infinispan.config.InfinispanProperties;
+import io.enmasse.iot.service.base.infinispan.devcon.DeviceConnectionKey;
+import io.enmasse.iot.service.base.infinispan.device.DeviceInformation;
+import io.enmasse.iot.service.base.infinispan.device.DeviceKey;
+import io.enmasse.iot.service.base.infinispan.tenant.TenantHandle;
+
+import io.vertx.config.ConfigRetriever;
+import io.vertx.config.ConfigRetrieverOptions;
+import io.vertx.config.ConfigStoreOptions;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonObject;
+
+import java.util.List;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.query.dsl.Expression;
+import org.infinispan.query.dsl.QueryBuilder;
+import org.infinispan.query.dsl.QueryFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InfinispanTenantCleaner {
+
+    private String tenantIdToClean;
+    private InfinispanProperties infinispanProperties;
+    private int maxQueryResults;
+
+    private RemoteCache<DeviceKey, DeviceInformation> devicesCache;
+    private RemoteCache<DeviceConnectionKey, String> devicesConnectionCache;
+
+    DeviceManagementCacheProvider mgmtProvider;
+    DeviceConnectionCacheProvider deviceconProvider;
+
+    private final Logger log = LoggerFactory.getLogger(InfinispanTenantCleaner.class);
+    private final Vertx vertx;
+
+    private final String pathToConfig;
+
+    public InfinispanTenantCleaner(String pathToConfig) {
+        VertxOptions vxOptions = new VertxOptions().setBlockedThreadCheckInterval(200000);
+        vertx = Vertx.vertx(vxOptions);
+        this.pathToConfig = pathToConfig;
+    }
+
+    public Future<Void> run(Future<Void> startPromise) {
```

Comment:
Most of the operations in this method will block. I think it is ok to implement this method classic, sync-style. And then invoke it with the "vertx blocking" wrapper. Otherwise it will only block the vertx main loop.

Reply:
@ctron  like this ? https://github.com/EnMasseProject/enmasse/pull/3154/commits/f859fb0e97ddef3fb2b63d9fbe1ee28a00f2d84b